nevergrad/functions/functionlib.py

```diff
@@ -63,6 +63,96 @@ def _short_repr(self) -> str:
+    """Population-Based Training, also known as Lamarckism or Meta-Optimization."""
+
+    def __init__(self, names: tp.List[str], dimensions: tp.List[int], num_workers: int):
+        for name in names:
+            if name not in corefuncs.registry:
+                available = ", ".join(sorted(corefuncs.registry))
+                raise ValueError(f'Unknown core function "{name}" in PBT. Available names are:\n-----\n{available}')
+        self._funcs = [corefuncs.registry[name] for name in names]
+        self._optima = [np.random.normal(size=d) for d in dimensions]
+        assert len(names) == len(dimensions)
+        self._dimension = len(names)
+        self._dimensions = dimensions
+        self._total_dimension = sum(dimensions)
+        parametrization = p.Array(shape=(self._dimension,)).set_name("")
+
+        # Population of checkpoints (that are optimized by the underlying optimization method)
+        # and parameters (that we do optimize).
+        self._population_checkpoints: tp.List[np.ndarray] = [np.zeros(self._total_dimension)] * num_workers
+        self._population_parameters: tp.List[np.ndarray] = [np.zeros(self._dimension)] * num_workers
+        self._population_fitness: tp.List[float] = [float("inf")] * num_workers
+        self._parameters = {x: y for x, y in locals().items() if x not in ["__class__", "self"]}
+        super().__init__(self.__func__, parametrization)
+        self.register_initialization(**self._parameters)
+
+    # The 3 methods below are function-specific.
+    def unflatten(self, x):
+        y = []
+        current_idx = 0
+        for i in range(self._dimension):
+            y += x[current_idx:(current_idx + self._dimensions[i])]
+            current_idx += self._dimensions[i]
+        assert current_idx == len(x) == self._total_dimension
+
+    def value(self, x):
+        y = self.unflatten(x)
+        return sum(f(xi - o) for f, xi, o in zip(self._funcs, y, self._optima))
+
+
+    def evolve(self, x: np.ndarray, p: np.ndarray):
+        assert len(p) == self._dimension
+        def gradient(f, x):
+            epsilon = 1e-15
+            # We compute a gradient by finite differences.
+            g = np.zeros(len(x))
+            for i in range(len(x)):
+                e = np.zeros(len(x))
+                e[i] = epsilon
+                g[i] = (f(x + e) - f(x)) / e
+            return g 
+        y = self.unflatten(x)
+        assert len(y) == self._dimension
+
+        for j in range(self._dimension):
+            assert len(y[j]) == self._dimensions[j]
+            y[j] -= np.exp(p[j]) * (gradient(self._funcs[j], y[j] - self._optima[j]) + np.random.normal(self._dimensions[j]))
+        current_idx = 0
+        for j in range(self._dimension):
+            x[current_idx:(current_idx) + self._dimensions[j]] = y[j]
+            current_idx += self._dimensions[j]
+        assert current_idx == self._total_dimension
+
+
+    def __func__(self, x: np.ndarray):
```

Comment:
never use this kind of double `_` names unless it is a special method, it's very dangerous in practice, you'd get weird behavior when subclassing. `_func` would be enough

Reply:
Done.